manager_tests/test_openstack.py

```diff
@@ -0,0 +1,294 @@
+            command = 'neutron net-show {0}'.format(resource_id)
+        elif 'Port' in resource_type:
+            command = 'neutron port-show {0}'.format(resource_id)
+        elif 'SecurityGroup' in resource_type:
+            command = 'neutron security-group-show {0}'.format(resource_id)
+        elif 'FloatingIP' in resource_type:
+            command = 'neutron floatingip-show {0}'.format(resource_id)
+        elif 'Server' in resource_type:
+            command = 'nova show {0}'.format(resource_id)
+        else:
+            raise Exception('Unsupported type {0} for {1}.'.format(
+                resource_type, resource_id))
+        self.assertEqual(0 if exists else 1, utils.execute_command(command))
+
+    def check_resources_in_deployment_created(self, nodes, node_names):
+        for node in nodes:
+            if node['id'] not in node_names:
+                break
+            external_id = \
+                node['instances'][0]['runtime_properties'].get('external_id')
+            self.check_openstack_resource(external_id, node['node_type'])
+
+    def check_resources_in_deployment_deleted(self, nodes, node_names):
+
+        for node in nodes:
+            if node['id'] not in node_names:
+                break
+            self.check_openstack_resource(
+                node['instances'][0]['runtime_properties']['external_id'],
+                node['node_type'], exists=False)
+
+    def install_manager(self):
+        self.manager_ip = labs.create_lab()
+
+    def uninstall_manager(self):
+        labs.delete_lab()
+
+    def upload_plugins(self):
+
+        sleep(5)
+        utils.update_plugin_yaml(
+            os.environ['CIRCLE_SHA1'], 'openstack')
+        workspace_path = os.path.join(
+            os.path.abspath('workspace'),
+            'build')
+        sleep(5)
+        utils.upload_plugin(utils.get_wagon_path(workspace_path))
+        sleep(5)
+        for plugin in self.plugins_to_upload:
+            utils.upload_plugin(plugin[0], plugin[1])
+
+    def setUp(self):
+
+        sensitive_data = [
+            os.environ['LAB_PASSWORD'],
+            os.environ['LAB_SERVER'],
+            os.environ['LAB_USERNAME']
+        ]
+        super(TestOpenstack, self).setUp(
+            'openstack.yaml', sensitive_data=sensitive_data)
+
+        os.environ['TEST_APPLICATION_PREFIX'] = \
+            os.environ['CIRCLE_BUILD_NUM']
+
+        if 'ECOSYSTEM_SESSION_MANAGER_IP' not in os.environ:
+            self.install_manager()
+            self.addCleanup(self.uninstall_manager)
+            self.password = 'admin'
+        os.environ['ECOSYSTEM_SESSION_PASSWORD'] = self.password
+        os.environ['ECOSYSTEM_SESSION_MANAGER_IP'] = self.manager_ip
+        utils.initialize_cfy_profile(
+            '{0} -u admin -p {1} -t default_tenant'.format(
+                self.manager_ip, self.password))
+        sleep(300)
+        self.upload_plugins()
+        self.add_os_environment_variables()
```

Comment:
Are we sure that all the secrets will be exists and whenever the first test run ? I mean do we expect to have an exception raised when retrieved secrets from manager

Reply:
The sleep 300 should wait until after everything is ready. I do think we need a better way of identifying manager readiness.